vitest.config.ts

```diff
@@ -0,0 +1,39 @@
+/// <reference types="vitest" />
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+import path from 'path'
+
+export default defineConfig({
+  plugins: [react()],
+  test: {
+    globals: true,
+    environment: 'happy-dom',
+    // setupFiles: ['./src/test/setup.ts'],
+    css: true,
+    // Include coverage configuration
+    coverage: {
+      provider: 'v8',
+      reporter: ['text', 'json', 'html'],
+      exclude: [
+        'node_modules/',
+        'src/test/',
+        '**/*.d.ts',
+        '**/*.config.*',
+        'src/vite-env.d.ts'
+      ]
+    },
```

Comment:
what are we doing with the coverage information? do we want to upload it somewhere?

Reply:
Currently nothing, a lot of the initial config was from trying to use AI to help me get something working that I eventually ditched. Sorry for forgetting to remove it from the PR.